File: redsheet-portal/src/app/database-models/v-lite-project-background-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vLiteProjectBackgroundRecord {
    constructor(vLiteProjectBackgroundRecord?: vLiteProjectBackgroundRecord )  {
        let properties = [ "dateOfNegotiation", "focus", "knowAboutThem", "knownIssues", "liteProjectBackgroundId", "liteRelationshipRequireId", "location", "opponentName", "projectModuleId", "reason", "valueObjectiveId",];
        for (let property of properties)
            if (vLiteProjectBackgroundRecord && vLiteProjectBackgroundRecord[property])
                this[property] = vLiteProjectBackgroundRecord[property];
    }
 
    @required()
	dateOfNegotiation : Date =   undefined;
 
    @required()
    @maxLength(100)
	focus : string =   undefined;
 
    @required()
    @maxLength(250)
	knowAboutThem : string =   undefined;
 
    @required()
    @maxLength(150)
	knownIssues : string =   undefined;
 
	liteProjectBackgroundId : number =   0 ;
 
    @range(1,2147483647)
	liteRelationshipRequireId : number =   undefined;
 
    @required()
    @maxLength(50)
	location : string =   undefined;
 
    @required()
    @maxLength(50)
	opponentName : string =   undefined;
 
    @range(1,2147483647)
	projectModuleId : number =   undefined;
 
    @required()
    @maxLength(100)
	reason : string =   undefined;
 
    @range(1,2147483647)
	valueObjectiveId : number =   undefined;


}

File: RedSheet.DbEntities/Models/vLiteProjectBackgroundRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vLiteProjectBackgroundRecords", Schema = "dbo")]
	[Serializable]
	public partial class vLiteProjectBackgroundRecord  
	{
 
	[Required]
	public System.DateTime DateOfNegotiation { get; set; } 
 
	[Required] 
	[MaxLength(100)]
	public string Focus { get; set; } 
 
	[Required] 
	[MaxLength(250)]
	public string KnowAboutThem { get; set; } 
 
	[Required] 
	[MaxLength(150)]
	public string KnownIssues { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int LiteProjectBackgroundId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int liteRelationshipRequireId { get; set; } 
 
	[Required] 
	[MaxLength(50)]
	public string location { get; set; } 
 
	[Required] 
	[MaxLength(50)]
	public string OpponentName { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectModuleId { get; set; } 
 
	[Required] 
	[MaxLength(100)]
	public string Reason { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ValueObjectiveId { get; set; } 

	public vLiteProjectBackgroundRecord()
	{
    }

    }
}

